x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -422,6 +440,14 @@ protected AllocatedPersistentTask createTask(
 
             String id = params.getId();
 
+            List<String> unavailableIndices = verifyIndicesPrimaryShardsAreActive(clusterState, AnomalyDetectorsIndex.configIndexName());
+            if (unavailableIndices.size() != 0) {
+                String reason = "Not opening job [" + id + "], because not all primary shards are active for the following indices [" +
```

Comment:
Do we want to use the word `job` here? Do you think we should say `data frame analytics job` and then the ID? Just to keep it clear it is not an anomaly job?

Reply:
Yes, good point.  The two types of jobs are in different namespaces so the message should say which it is.  I'll change the equivalent message in the other file too.

Reply:
> I'll change the equivalent message in the other file too.

On reflection I think changing "job" to "anomaly detection job" in existing error messages should be done in a different PR.  There are many places that would need changing to make it consistent.